packages/components/src/internal/components/samples/SampleCreationTypeModal.tsx

```diff
@@ -30,33 +33,42 @@ export class SampleCreationTypeModal extends React.PureComponent<Props, State> {
             numPerParent: 1,
             submitting: false,
         };
+        this._maxPerParent = MAX_EDITABLE_GRID_ROWS / props.parentCount;
     }
 
     onCancel = () => {
         this.props.onCancel();
     };
 
     onChange = event => {
-        const { name, value } = event.target;
-        this.setState({ [name]: value } as State);
+        const {name, value} = event.target;
+        this.setState({[name]: value} as State);
     };
 
+    isValidNumPerParent() {
+        const {numPerParent} = this.state;
+
+        return numPerParent >= 1 && numPerParent <= this._maxPerParent;
```

Comment:
I thought I remembered there being an issue opened on this. If so, do you want to add that issue number in these changes?